tests/test_record_replay.py

```diff
@@ -154,3 +154,44 @@ def test_init_and_rec(self):
         assert names[0].endswith('.warcgz')
 
 
+# ============================================================================
+class TestRecordFilter(HttpBinLiveTests, CollsDirMixin, BaseConfigTest):
+
+    @classmethod
+    def setup_class(cls):
+        rec_custom = {'collections': {'fallback': {'sequence': [
+                        {
+                            'index_paths': os.path.join(TEST_CDX_PATH, 'example.cdxj'),
+                            'archive_paths': TEST_WARC_PATH,
+                            'name': 'example'
+                        },{
+                            'index':'$live',
+                            'name': 'live'
+                        }]}},
+                        'recorder': {'source_coll': 'fallback',
+                                     'source_filter': 'live',
+                                     }
+                     }
+        super(TestRecordFilter, cls).setup_class('config_test_record.yaml', custom_config=rec_custom)
+
+    def test_init_and_rec(self):
+        manager(['init', 'test-new'])
+        dir_name = os.path.join(self.root_dir, '_test_colls', 'test-new', 'archive')
+        assert os.path.isdir(dir_name)
+        res = self.testapp.get('/fallback/cdx?url=http://example.com/?example=1')
+        assert res.text != ''
+        res = self.testapp.get('/fallback/cdx?url=http://httpbin.org/get?A=B')
+        assert res.text == ''
+
+        res = self.testapp.get('/test-new/record/mp_/http://example.com/?example=1')
```

Comment:
Very minor change, could you split these two tests into separate functions, maybe `test_filtered_record_not_recorded` and `test_filtered_record_recorded` to indicate that first one is not recorded as its coming from example collection, and second one is recorded as its coming from live?